grpc-client-utils/src/main/java/org/hypertrace/core/grpcutils/client/GrpcChannelRegistry.java

```diff
@@ -0,0 +1,34 @@
+package org.hypertrace.core.grpcutils.client;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GrpcChannelRegistry {
+  private static final Logger LOG = LoggerFactory.getLogger(GrpcChannelRegistry.class);
+  private final Map<String, ManagedChannel> channelMap = new ConcurrentHashMap<>();
+  private volatile boolean isShutdown = false;
```

Comment:
Removed the integrated shutdown on this one so it didn't bring in more dependencies (the shutdown lifecycle is from), but I can bring it back if that seems more useful. The more I think on it, the more I think I'll optionally add just the future part of it (to avoid the dependency)

Reply:
Changed my mind again - the difference of accepting a hook is really just the below anyway:
```
new GrpcChannelRegistry(future);
// vs
registry = new GrpcChannelRegistry();
future.thenRun(registry::shutdown);
```